source/sparse_matrix_simd.template.h

```diff
@@ -224,30 +224,26 @@ namespace ryujin
             continue;
 
           const unsigned int row = *idx;
-          indices_to_be_sent.push_back(row < n_internal_dofs
-                                           ? row_starts[row / simd_length] +
-                                                 row % simd_length
-                                           : row_starts[row]);
+
+          entries_to_be_sent.push_back({row, 0});
 
           for (auto jt = ++sparsity.begin(row); jt != sparsity.end(row); ++jt) {
             if (jt->column() >= ghost_ranges[p_match] &&
                 jt->column() < ghost_ranges[p_match + 1]) {
-              indices_to_be_sent.push_back(
-                  row < n_internal_dofs
-                      ? row_starts[row / simd_length] + row % simd_length +
-                            (jt - sparsity.begin(row)) * simd_length
-                      : row_starts[row] + (jt - sparsity.begin(row)));
+              const unsigned int position_within_column =
+                  jt - sparsity.begin(row);
+              entries_to_be_sent.push_back({row, position_within_column});
```

Comment:
I changed the `indices_to_be_sent` vector to an `entries_to_be_sent` vector storing a pair `{row, position_within_column}` instead of an absolute (serial) index into `data`.